framework/src/application/node/transport/transport.js

```diff
@@ -398,6 +394,10 @@ class Transport {
 			throw err;
 		}
 
+		if (this.transactionPoolModule.contains(transactionJSON.id)) {
+			return transactionJSON.id;
```

Comment:
```suggestion
			return transaction.id;
```
since `transactionJSON` might not contain `id`, it should use `transaction`